src/modules/emit_sound.rs

```diff
@@ -0,0 +1,219 @@
+        static COMMANDS: &[&Command] = &[
+            &BXT_EMIT_SOUND_STOP_SOUND_EXCEPT,
+            &BXT_EMIT_SOUND_DYNAMIC,
+            &BXT_EMIT_SOUND,
+        ];
+        COMMANDS
+    }
+
+    fn is_enabled(&self, marker: MainThreadMarker) -> bool {
+        engine::SV_StartSound.is_set(marker)
+            && engine::S_StartDynamicSound.is_set(marker)
+            && engine::S_StopSound.is_set(marker)
+            && engine::S_PrecacheSound.is_set(marker)
+    }
+}
+
+#[derive(Default)]
+struct SoundInfo {
+    sound: String,
+    channel: i32,
+    /// Volume is [0..1]
+    volume: f32,
+    from: i32,
+    to: i32,
+    /// This is the "radius" of the sound. Close to 1, you hear sound quieter when moving farther
+    /// from entity. "NORM" for sound-emitting entity is 0.8.
+    /// 0 is not quite to make it "global". Negative value will do.
+    attenuation: f32,
+    flag: i32,
+    /// 100 is no shift.
+    pitch: i32,
+}
+
+impl SoundInfo {
+    fn new(sound: String, channel: i32) -> Self {
+        SoundInfo {
+            sound,
+            channel,
+            volume: 1.,
+            from: 0,
+            to: 0,
+            attenuation: 0.,
+            flag: 0,
+            pitch: 100,
+        }
+    }
+}
+
+// Eh.
+// Parse float then round then convert to integer to avoid a more convoluted solution.
+impl FromStr for SoundInfo {
+    type Err = ParseFloatError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut rv = SoundInfo::default();
+
+        let mut iter = s.split_ascii_whitespace();
+        rv.sound = iter.next().unwrap_or_default().to_string();
+        rv.channel = (f32::from_str(iter.next().unwrap_or_default())?).round() as i32;
+        rv.volume = f32::from_str(iter.next().unwrap_or_default())?;
+        rv.from = (f32::from_str(iter.next().unwrap_or_default())?).round() as i32;
+        rv.to = (f32::from_str(iter.next().unwrap_or_default())?).round() as i32;
+        rv.attenuation = f32::from_str(iter.next().unwrap_or_default())?;
+        rv.flag = (f32::from_str(iter.next().unwrap_or_default())?).round() as i32;
+        rv.pitch = (f32::from_str(iter.next().unwrap_or_default())?).round() as i32;
+
+        Ok(rv)
+    }
+}
+
+static BXT_EMIT_SOUND: Command = Command::new(
+    b"bxt_emit_sound\0",
+    handler!(
+        "bxt_emit_sound <sound> <channel> [volume] [from] [to] [attenuation] [flag] [pitch]
+Plays sound file directly from SV_StartSound.",
+        emit_sound as fn(_, _, _),
+        emit_sound_full as fn(_, _)
+    ),
+);
+
+fn emit_sound(marker: MainThreadMarker, sound: String, channel: i32) {
+    emit_sound_full(marker, SoundInfo::new(sound, channel));
+}
+
+fn emit_sound_full(marker: MainThreadMarker, info: SoundInfo) {
+    if let Some(player) = unsafe { engine::player_edict(marker) } {
+        let mut entity = unsafe { NonNull::new(player.as_ptr().add(info.from as usize)).unwrap() };
+
+        let entity = unsafe { entity.as_mut() };
```

Comment:
That is a bit of a roundabout way. I think you can do without the intermediate NonNull.